htcanalyze/main.py

```diff
@@ -670,7 +686,7 @@ def run(commandline_args):
 
         logging.debug("-------Start of htcanalyze script-------")
 
-        show_legend = not GlobalServant.redirecting_stdout  # redirected ?
+        show_legend = not GlobalStream.redirecting_stdout  # redirected ?
```

Comment:
`# redirected ?`: Unsure what this means in this context.

Reply:
Apparently what this does is take the default of `None` (which isn't changed anywhere) and take the opposite. Since `None` is `False`, this will return `True`. Hence, if `GlobalStream.redirecting_stdout` is `None`, it will show the legend.

I'm not a fan of `GlobalStream` (or `GlobalPlayer`). Instead of creating a class and using its methods, mathis can instead just check `sys.stdin.isatty()`, `sys.stdin.readlines()`, and `sys.stdout.isatty()`.